### PR TITLE
Fix Exception misuse in `ignite.contrib.handlers.param_scheduler.py`

### PR DESCRIPTION
**Pull request by WrRan**
Fixes # 

Description:
1. `TypeError` may be raised when an operation or function is applied to an object of an inappropriate type, while `RuntimeError` may be raised when an error is detected that doesn’t fall in any of the other categories.
2. `TypeError` may be raised when an operation or function is applied to an object of an inappropriate type, while `ValueError` may be raised when an operation or function receives an argument that has the right type but an inappropriate value.

see python at https://docs.python.org/3/library/exceptions.html

Check list:
* [ ] New tests are added (if a new feature is added)
* [x] New doc strings: description and/or example code are in RST format
* [x] Documentation is updated (if required)

### CODE REVIEW

File: ignite/contrib/handlers/param_scheduler.py [hunk 1]
```diff
@@ -567,8 +573,15 @@ def simulate_values(cls, num_events, schedulers, durations, param_names=None, **
             list of [event_index, value_0, value_1, ...], where values correspond to `param_names`.
 
         """
-        if param_names is not None and not isinstance(param_names, (list, tuple)):
-            raise ValueError("Argument param_names should be list or tuple of strings")
+        if param_names is not None:
+            if not isinstance(param_names, (list, tuple)):
+                raise TypeError(
+                    "Argument param_names should be list or tuple, " "but given {}".format(type(param_names))
```
Comment (vfdev-5) [inline ignite/contrib/handlers/param_scheduler.py:579]:
```suggestion
                    "Argument param_names should be list or tuple, but given {}".format(type(param_names))
```

File: ignite/contrib/handlers/param_scheduler.py [hunk 2]
```diff
@@ -567,8 +573,15 @@ def simulate_values(cls, num_events, schedulers, durations, param_names=None, **
...
+                )
+            if not all(isinstance(item, str) for item in param_names):
+                raise ValueError(
+                    "Argument param_names should be list or tuple of strings, " "but given {}".format(param_names)
```
Comment (vfdev-5) [inline ignite/contrib/handlers/param_scheduler.py:583]:
```suggestion
                    "Argument param_names should be list or tuple of strings, but given {}".format(param_names)
```

File: ignite/contrib/handlers/param_scheduler.py [hunk 3]
```diff
@@ -857,10 +873,14 @@ class PiecewiseLinear(ParamScheduler):
     def __init__(self, optimizer, param_name, milestones_values, save_history=False, param_group_index=None):
         super(PiecewiseLinear, self).__init__(optimizer, param_name, save_history, param_group_index=param_group_index)
 
-        if not isinstance(milestones_values, Sequence) or len(milestones_values) < 1:
+        if not isinstance(milestones_values, Sequence):
+            raise TypeError(
+                "Argument milestones_values should be a list or tuple, " "but given {}".format(type(milestones_values))
```
Comment (vfdev-5) [inline ignite/contrib/handlers/param_scheduler.py:878]:
```suggestion
                "Argument milestones_values should be a list or tuple, but given {}".format(type(milestones_values))
```

File: ignite/contrib/handlers/param_scheduler.py [hunk 4]
```diff
@@ -938,16 +958,24 @@ class ParamGroupScheduler(ParamScheduler):
...
-        if not (
-            isinstance(schedulers, Sequence) and all(isinstance(scheduler, ParamScheduler) for scheduler in schedulers)
-        ):
-            raise ValueError("Argument schedulers should be a list/tuple of parameter schedulers")
+        if not isinstance(schedulers, Sequence):
+            raise TypeError("Argument schedulers should be a list/tuple, but given {}".format(schedulers))
+
+        if not all(isinstance(scheduler, ParamScheduler) for scheduler in schedulers):
+            raise ValueError(
+                "Argument schedulers should be a list/tuple of parameter schedulers, " "but given {}".format(schedulers)
```
Comment (vfdev-5) [inline ignite/contrib/handlers/param_scheduler.py:966]:
```suggestion
                "Argument schedulers should be a list/tuple of parameter schedulers, but given {}".format(schedulers)
```

File: ignite/contrib/handlers/param_scheduler.py [hunk 5]
```diff
@@ -938,16 +958,24 @@ class ParamGroupScheduler(ParamScheduler):
...
             names = [s.param_name for s in schedulers]
 
-        if not (isinstance(names, (list, tuple)) and all(isinstance(n, str) for n in names)):
-            raise ValueError("Argument names should be a list/tuple of parameter scheduler's names")
+        if not isinstance(names, (list, tuple)):
+            raise TypeError("Argument names should be a list/tuple, but given {}".format(names))
+
+        if not all(isinstance(n, str) for n in names):
+            raise ValueError(
+                "Argument names should be a list/tuple of parameter scheduler's names, " "but given {}".format(names)
```
Comment (vfdev-5) [inline ignite/contrib/handlers/param_scheduler.py:977]:
```suggestion
                "Argument names should be a list/tuple of parameter scheduler's names, but given {}".format(names)
```

File: tests/ignite/contrib/handlers/test_param_scheduler.py [hunk 6]
```diff
@@ -317,10 +317,10 @@ def test_concat_scheduler_asserts():
     with pytest.raises(ValueError):
         ConcatScheduler(schedulers=[scheduler_1, scheduler_2, scheduler_2], durations=[15, 12.0])
 
-    with pytest.raises(ValueError):
+    with pytest.raises(TypeError):
```
Comment (vfdev-5) [inline tests/ignite/contrib/handlers/test_param_scheduler.py:320]: Could you please add `, match=r"...message..."` to all those `pytest.raises`.

**Comment by WrRan** [comment]
Got it. I am working on it. @vfdev-5

**Comment by vfdev-5** [comment]
@WrRan any updates on this ?

**Comment by WrRan** [comment]
@vfdev-5 Please have a look.

**Pull request merged**